tobac/feature_detection.py

```diff
@@ -1463,12 +1525,35 @@ def feature_detection_multithreshold(
                 )
             features = pd.concat(filtered_features, ignore_index=True)
 
+        # we map the feature index to the original index
+        if return_labels:
+
+            for i, time_i, label_field_i, features_i in field_and_features_over_time(
+                label_fields, features
+            ):
+                wh_all_labels = np.isin(label_field_i, features_i.idx)
+
+                remapper = xr.DataArray(
+                    features_i.feature, dims=("idx",), coords=dict(idx=features_i.idx)
+                )
+
+                label_fields[i].data[wh_all_labels] = remapper.loc[
+                    label_field_i.data[wh_all_labels]
+                ]
+                label_fields[i].data[~wh_all_labels] = 0
+
     else:
         features = None
+        label_fields = None
         logging.debug("No features detected")
 
     logging.debug("feature detection completed")
-    return features
+
+    # Create the final output
+    if return_labels:
+        return label_fields, features
```

Comment:
I would recommend changing the order of the returned items, so that the labels come after the dataframe as is common for optional return items in other libraries e.g. numpy. This might require changes elsewhere to the type hints and docstring
```suggestion
        return features, label_fields
```